### PR TITLE
SI-7498 ParTrieMap.foreach no longer crashes

### PR DESCRIPTION
**Pull request by axel22**
Previously, the `split` method of the `ParTrieMap` iterator threw
an exception when splitting a splitter that iterated over nodes
whose hash codes collide.
This was due to reusing the iterator of the list of colliding keys
rather than creating a new splitter.

This commit changes the `subdivide` method to create a new
iterator using the factory method of the current trie map
iterator rather than returning a `LinearSeqLike` iterator.

Review by @retronym

### CODE REVIEW

**Comment by JamesIry** [comment]
Could you redo your commit message to follow https://github.com/scala/scala/wiki/Pull-Request-Policy ?

**Comment by axel22** [comment]
Absolutely, hope it's ok now.

**Comment by JamesIry** [comment]
@axel22 Thanks, 100x better! But the commit message still needs fixing. :-)

**Comment by axel22** [comment]
@JamesIry I shortened the text in the title, to below 50 characters.
I think all the lines should be below 72 characters now.
Would be helpful if we had automatic character counts here.

Could you be more specific as to where it doesn't satisfy the guidelines? :)
(It's not entirely in imperative/present tense, but I noticed that other PRs aren't either when describing what was failing previously)

**Comment by axel22** [comment]
Ah, sorry - you were referring to the commit message, not the PR one. Will fix now.

File: test/files/run/t7498.scala [hunk 1]
```diff
@@ -0,0 +1,20 @@
+
+
+
+
+
+
+
```
Comment (retronym) [inline test/files/run/t7498.scala:7]: I always wonder if I have a white-font-on-white-background when I hit your patches of "negative space", and wonder if I'm missing some cool code underneath. I'll merge this change regardless, but will gently suggest a more compact style in the future :)

**Comment by retronym** [comment]
LGTM (with the caveat that I haven't dived into ParTrieMap internals, so I'm trusting your well documented assessment in the commit message and the test case.)

Do you think the same class of bug could affect other parts of the implementation?

File: test/files/run/t7498.scala (hunk 1 above)
Comment (axel22) [inline test/files/run/t7498.scala:7]: Ok. Actually, no cool code underneath, I guess I just like my code in the middle of the screen :)
Will not include the negative space in the future, and perhaps write a Sublime plugin to render extra space for short files.

**Comment by axel22** [comment]
I don't think there should be a similar class of bugs, at least I am not aware of it, but if any arise, I'd be more than glad to fix them in the future.

**Pull request merged**